### PR TITLE
Support categorical_partitions argument in dask-cudf read_parquet

### PR DESCRIPTION
**Pull request by rjzamora**
Blocked by [dask#8050](https://github.com/dask/dask/pull/8050)

This PR exposes `read_parquet(..., categorical_partitions=False)` in Dask-cuDF, which was introduced upstream in [dask#8050](https://github.com/dask/dask/pull/8050).

### CODE REVIEW

**Comment by quasiben** [comment]
rerun tests

**Comment by quasiben** [comment]
Thanks for adding this @rjzamora .  cc @randerzander

File: python/dask_cudf/dask_cudf/io/parquet.py [hunk 1]
```diff
@@ -142,23 +142,32 @@ def read_partition(
...
+                    )
+                    df[name] = build_categorical_column(
+                        categories=partitions[i].keys,
+                        codes=codes,
+                        size=codes.size,
+                        offset=codes.offset,
+                        ordered=False,
+                    )
+                else:
+                    # Repeat index2 for every element of the new column
```
Comment (quasiben) [inline python/dask_cudf/dask_cudf/io/parquet.py:164]: do you think it's worth adding a note that round tripping dtypes won't work.   For example, Int64 columns can be int32 after reading

**Comment by vyasr** [comment]
@rjzamora is this still something that we want?

**Pull request closed without merging**